src/components/root/createModals/CreateScriptModal.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+	Button,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	Divider,
+	FormControl,
+	FormHelperText,
+	FormLabel,
+	Input,
+	Modal,
+	ModalDialog,
+} from '@mui/joy';
+import { CreateModalsProps } from './createModalsProps';
+import { iconFromTabType, Script } from '../../../types/application-data/application-data';
+import { useEffect, useMemo, useState } from 'react';
+import { toValidFunctionName } from '../../../utils/string';
+import { InfoOutlined } from '@mui/icons-material';
+import { useAppDispatch } from '../../../state/store';
+import { addTabs, setSelectedTab } from '../../../state/tabs/slice';
+import { createScript } from '../../../state/active/thunks/scripts';
+
+export function CreateScriptModal({ open, closeFunc }: CreateModalsProps) {
+	const [scriptName, setScriptName] = useState('');
+	const [scriptCallingName, setScriptCallingName] = useState('');
+	const dispatch = useAppDispatch();
+	const createScriptFunction = async () => {
+		const newScript: Partial<Script> = { name: scriptName, scriptCallableName: scriptCallingName, content: '' };
+
+		const createdScriptId = await dispatch(createScript(newScript)).unwrap();
+		dispatch(addTabs({ [createdScriptId]: 'script' }));
+		dispatch(setSelectedTab(createdScriptId));
+	};
+	const scriptCallingNameValid =
+		scriptCallingName.length > 0 && toValidFunctionName(scriptCallingName) === scriptCallingName;
+	const scriptNameValid = scriptName.length > 0;
+	const allFieldsValid = useMemo(() => scriptNameValid && scriptCallingNameValid, [scriptName, scriptCallingName]);
```

Comment:
I think this is another instance of a useMemo that is probably doing more harm than good. You've already done all the work, so you're just memoizing a single &&